images/bootstrap/runner.sh

```diff
@@ -41,77 +41,59 @@ setup_ca(){
-        docker ps -aq | xargs -r docker rm -f || true
-        kill "$(</var/run/docker.pid)" || true
-        wait "$(</var/run/docker.pid)" || true
+# runs custom podman data root cleanup binary and debugs remaining resources
+cleanup_pinc() {
+    if [[ "${PODMAN_IN_CONTAINER_ENABLED:-false}" == "true" ]]; then
+        echo "Cleaning up after podman"
+        podman ps -aq | xargs -r podman rm -f || true
+        kill "$(</var/run/podman.pid)" || true
+        wait "$(</var/run/podman.pid)" || true
     fi
 }
 
 early_exit_handler() {
-    cleanup_dind
+    cleanup_pinc
 }
 
 # setup certificates before anything gets started
 setup_ca
 
-# optionally enable ipv6 docker
-export DOCKER_IN_DOCKER_IPV6_ENABLED=${DOCKER_IN_DOCKER_IPV6_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_IPV6_ENABLED}" == "true" ]]; then
-    echo "Enabling IPV6 for Docker."
-    # configure the daemon with ipv6
-    mkdir -p /etc/docker/
-    cat <<EOF >/etc/docker/daemon.json
-{
-  "ipv6": true,
-  "fixed-cidr-v6": "fc00:db8:1::/64"
-}
-EOF
+# optionally enable ipv6
+export PODMAN_IN_CONTAINER_IPV6_ENABLED=${PODMAN_IN_CONTAINER_IPV6_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_IPV6_ENABLED}" == "true" ]]; then
+    echo "Enabling IPV6."
     # enable ipv6
     sysctl net.ipv6.conf.all.disable_ipv6=0
     sysctl net.ipv6.conf.all.forwarding=1
     # enable ipv6 iptables
     modprobe -v ip6table_nat
 fi
 
-# Check if the job has opted-in to docker-in-docker availability.
-export DOCKER_IN_DOCKER_ENABLED=${DOCKER_IN_DOCKER_ENABLED:-false}
-if [[ "${DOCKER_IN_DOCKER_ENABLED}" == "true" ]]; then
-    echo "Docker in Docker enabled, initializing..."
-    printf '=%.0s' {1..80}; echo
-    # If we have opted in to docker in docker, start the docker daemon,
-    (
-        if [ -f "/etc/default/docker" ]; then
-            source /etc/default/docker
-        fi
-        /usr/bin/dockerd \
-            -p /var/run/docker.pid \
-            --data-root=/docker-graph \
-            --init-path /usr/libexec/docker/docker-init \
-            --userland-proxy-path /usr/libexec/docker/docker-proxy \
-            ${DOCKER_OPTS} \
-                >/var/log/dockerd.log 2>&1 &
-    )
-    # the service can be started but the docker socket not ready, wait for ready
+export PODMAN_IN_CONTAINER_ENABLED=${PODMAN_IN_CONTAINER_ENABLED:-false}
+if [[ "${PODMAN_IN_CONTAINER_ENABLED}" == "true" ]]; then
+    echo "Podman in Container enabled, initializing in podman compatible mode..."
+    podman system service \
+           -t 0 \
+           unix:///${HOME}/podman.sock \
+           >/var/log/podman.log 2>&1 &
+    echo "${!}" > /var/run/podman.pid
+    # the service can be started but the socket not ready, wait for ready
     WAIT_N=0
     MAX_WAIT=5
     while true; do
-        # docker ps -q should only work if the daemon is ready
-        docker ps -q > /dev/null 2>&1 && break
+        # podman ps -q should only work if the daemon is ready
+        podman ps -q > /dev/null 2>&1 && break
```

Comment:
Do you think we should symlink `podman` and `docker` to make the transition for users easier?